Works & Royalty Interest Representation (#5553)
ethereum/EIPs: EIPS/eip-5553.md

---
eip: 5553
title: Works & Royalty Interest Representation
description: A way of representing intellectual IP works and their respective royalty interests on chain
author: Roy Osherove (@royosherove)
discussions-to: https://ethereum-magicians.org/t/eip-821-simple-works-royalty-interest-representation-standard/10551
status: Draft
type: Standards Track
category: ERC
created: 2022-08-17
requires: 20, 721
---

> ## Abstract

Reviewer: [Editorial] It seems there are 3 subsections of Abstract. I wonder if it would be too long for what would be considered as "Abstract". Could you shorten the Abstract and then maybe everything else could be merge into Motivation or Specification?

Author: It's now shorter.

> A standardized way to represent intellectual works on chain, along with a refined royalty representation mechanism and associated metadata. This standard is not associated with a specific type of work and could represent many types of works such as musical works, videos, books, images and more.

Reviewer 2: > A standardized way to represent intellectual works on chain, along with a refined royalty representation mechanism and associated metadata. 

Please make this a complete sentence.

The standard is kept very generic on purpose to allow the industry to evolve new ecosystems that can all rely on the same basic standard at their core.

This standard allows market participants to:
1) Observe the canonical on-chain representation of an intellectual work
2) Discover its attached metadata
3) Discover its related royalty rights holders
4) In the future, this will enable building registration, licensing and payout mechanisms for intellectual property assets.

> The standard is based on an `EIP-721` based contract implementing the new `IWorksRegistration` interface.

Reviewer: [Editorial] We start referring to EIPs in the format of `[EIP-NNN(./eip-NNN.md)]`. Please update.

There is no accepted standard mechanism to license a work, or to represent it, except using traditional NFTs. But regular NFTs only represent a collectible item use case, and cannot easily represent more complicated use cases of licensing a work for different types of uses.

> To enable such mechanisms, we need a more robust mechanism for :

Reviewer 2: ```suggestion
To enable such mechanisms, a more robust mechanism is needed to:
```

> 1) Declaring that a work exists, SEPARATE from its purchase ability
> 2) Declaring possibly multiple interested parties to be paid for such work 

Reviewer 2: ```suggestion
1) Declare that a work exists, SEPARATELY from its purchase ability
2) Declare possibly multiple interested parties to be paid for such work 
```

> returns an array of addresses related to [EIP-20](./eip-20.md) tokens that represent royalty interest allocation to different types of interested parties that expect future royalty distribution. These royalty tokens represent a more granular and streamlined way to declare royalty splits for multiple collaboration participants for the creation of the work. For example, for a musical work, we might have two tokens representing the Composition/Writing/Publishing royalty rights side, and the Recording/Master side. 

Reviewer: [Editorial]Move to the spec?

> for #1 no standard exists today.
> For #2 we only have regular splits standards based on NFT purchases, or through mechanisms like 0xsplits. While these are a great headstart, they do not contain the ability to name multiple types of collaboration participants, and more importantly, do not contain an easy way to manage the splits in a way that feels natural.

Reviewer 2: ```suggestion
For 1, no standard exists today.

For 2, there are regular split standards based on NFT purchases, or through mechanisms like 0xsplits. While these solve the main problem, they do not contain the ability to name multiple types of collaboration participants.
```

> With this proposal, registration is taken care of, and splits are taken care of by defining the splits as an open array of [EIP-20](./eip-20.md) tokens that represent split percentages. Moving splits to various participants will be as simple as sending an EIP-20 token from one wallet to another. discovering split movements is as simple as checking [EIP-20](./eip-20.md) transfer events.
> 
> In the future, we can build full payout mechanisms based on these simple split tokens, and licensing standards on top of those payout mechanisms, which are not possible today due to the complexity of splits management in the traditional web2 world.

Reviewer 2: ```suggestion
```
This is abstract / rationale

## Specification 

> The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
> NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and
> "OPTIONAL" in this document are to be interpreted as described in
> RFC 2119.

Reviewer 2: ```suggestion
The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
```

**contracts that want to represent a work on chain MUST implement [EIP-721](./eip-721.md) AND this Proposal**

> The standard is based on an [EIP-721](./eip-721.md) based contract implementing the new `IWorksRegistration` interface.

Reviewer 2: ```suggestion
This standard extends [EIP-721](./eip-721.md) with the following `IWorksRegistration` interface.
```

> There is no accepted standard mechanism to license a work, or to represent it, except using traditional NFTs. But regular NFTs only represent a collectible item use case, and cannot easily represent more complicated use cases of licensing a work for different types of uses.

Reviewer: [Peer review] Did you get a chance to check out EIP-5218? Do you think 5218's relevant to this EIP?

Author: I've seen it before  - but had not considered connecting the dots. I might tryu to mention here in this context as a possible use case along with works registration/

Implementers of this standard **MUST** have all of the following functions:

### royaltyInterestTokens() function

> returns an array of addresses related to [EIP-20](./eip-20.md) tokens that represent royalty interest allocation to different types of interested parties that expect future royalty distribution. These royalty tokens represent a more granular and streamlined way to declare royalty splits for multiple collaboration participants for the creation of the work. For example, for a musical work, we might have two tokens representing the Composition/Writing/Publishing royalty rights side, and the Recording/Master side. 

Reviewer 2: ```suggestion
This function an array of addresses related to [EIP-20](./eip-20.md) tokens that represent royalty interest allocation to different types of interested parties that expect future royalty distribution. These royalty tokens represent a more granular and streamlined way to declare royalty splits for multiple collaboration participants for the creation of the work. For example, for a musical work, we might have two tokens representing the Composition/Writing/Publishing royalty rights side, and the Recording/Master side. 
```

These royalty tokens are distributed to the collaboration participants and can later be queried for the various holders, for purposes of distribution of royalties - i.e if you hold 10% of a royalty token, you might get 10% of financial distribution related to that type of royalty.

### metadataURI() function

> returns the URI to a metadata file containing any required metadata for the work. Each work type will have its own metadata standard, defined separately, and is not the focus of this standard. The hope is that the file will be hosted in IPFS or arweave or other resource hash based access systems in which we know the file's contents are not changeable without changing the URI.

Reviewer 2: ```suggestion
This function returns the URI to a metadata file containing any required metadata for the work. Each work type will have its own metadata standard, defined separately, and is not the focus of this standard. The hope is that the file will be hosted in IPFS or Arweave or other resource hash-based access systems in which the file's contents are not changeable without changing the URI.
```

### changeMetadataURI() function
Allows changing the URI when updating the metadata for the work. Will also trigger the event `MetadataChanged`

### ledger() function

> returns the address of the registry or registrar contract, or an EOA account that initialized the work and associated royalty tokens. A work might be registered in multiple places by different actors for different purposes, and this enables market participants to discover which registry mechanism is the parent of the work and might have special access rights to manage the work.

Reviewer 2: ```suggestion
This function returns the address of the registry or registrar contract, or an EOA account that initialized the work and associated royalty tokens. A work might be registered in multiple places by different actors for different purposes, and this enables market participants to discover which registry mechanism is the parent of the work and might have special access rights to manage the work.
```

```solidity
// SPDX-License-Identifier: Apache-2.0
pragma solidity ^0.8.9;
import '@openzeppelin/contracts/interfaces/IERC165.sol';

///
/// @dev Interface for Simple Works Registration Standard
///
interface IWorksRegistration is IERC165 {
    
    /// @notice Called with the new uri to an updated metadata file
    /// @param _newUri - the uri pointing to a metadata file (file standard is up to the implementer)
    /// @param _newFileHash - The hash of the new metadata file, for future reference and verification

>     function changeMetadataURI(string memory _newUri, string memory _newFileHash) external ;

Reviewer: [Editorial, Peer] Consider add an event when URL changed?

Author: There is already a MetadataChanged event specified:
    event MetadaDataChanged(address byAddress,string oldURI, string oldFileHash, string newURI, string newFileHash);

    /// @return array of addresses of ERC20 tokens representing royalty interest in the work
    /// @dev i.e implementing ERC5501 (IRoyaltyInterestToken interface)
    function royaltyInterestTokens() external view returns (address[] memory) ;

    /// @return the address of the contract or EOA that initialized the work registration
    /// @dev i.e a registry or registrar, to be implemented in the future
    function ledger() external view returns (address) ;

    /// @return the uri of the current metadata file for the work
    function metadataURI() external view returns (string memory) ;

    /// @dev event to be triggered whenever metadata URI is changed
    /// @param byAddress the addresses that triggered this operation
    /// @param oldURI the uri to the old metadata file prior to the change
    /// @param oldFileHash the hash of the old metadata file prior to the change
    /// @param newURI the uri to the new metadata file 
    /// @param newFileHash the hash of the new metadata file 
    event MetadaDataChanged(address byAddress,string oldURI, string oldFileHash, string newURI, string newFileHash);
}
```

> ### Examples 

Reviewer: [Editorial] Move to "Reference Implementation" section?

Author: Renamed.

> ### Reference Implementation 

Reviewer 2: ```suggestion
## Reference Implementation 
```

import "@openzeppelin/contracts/utils/Counters.sol";
import "./SongRegistration.sol";
import "./CompositionRoyaltyToken.sol";
import "./RecordingRoyaltyToken.sol";

contract SimpleSongLedger is IERC721Receiver {
    using Counters for Counters.Counter;
    Counters.Counter private songIds;
      function onERC721Received(address, address, uint256, bytes calldata) external pure returns (bytes4) {
        return IERC721Receiver.onERC721Received.selector;
    }

    function mintSong(SongMintingParams memory _params) public {
        CompositionRoyaltyToken comp = new CompositionRoyaltyToken(address(this),"SONGCOMP","COMP");
        RecordingRoyaltyToken rec = new RecordingRoyaltyToken(address(this),"SONGREC","REC");
        songIds.increment();

        SongRegistration newSong = new SongRegistration(
                                        songIds.current(),
                                        address(this),
                                        _params,
                                        address(comp),
                                        address(rec)
                                    );
    }
}

```

## Rationale

### A more robust royalty interest structure

> Current royalty implementations deal only with a single type of royalty payments: NFT sales. They also only allow a single type of royalty - i.e you cannot have different royalty interest owners receive royalty based on different licensing types. 

Reviewer: Rationale explains how design decisions are made, motivation explains why author proposes this EIP and why having such EIP is better than not having one. I feel this 's more aligned with motivation

Author: Fixed working and moved one subsection to motivation.

In other words, currently, a royalty split works the same way no matter what type of purchase or license deal has happened, for all parties involved.

> With this proposal, we allow multiple **types** of royalty interests. A classic case is the music industry in which you have writing/composition royalties, and recording/master royalties. Different licensing types will pay different percentages to different parties based on context.

Reviewer 2: ```suggestion
With this proposal, multiple **types** of royalty interests are allowed. A classic case is the music industry in which you have writing/composition royalties, and recording/master royalties. Different licensing types will pay different percentages to different parties based on context.
```

In the case of a song cover, a license payment formula can be created so that that 
a) Original work's writers get paid for using the lyrics or composition of the song
b) recording artists of the original song do not get paid since their recording is not used
c) recording artists of the new work will get paid
d) there are no writing royalties for the creators of the cover.

> Moreover, we have single structure that connects to all types of royalty types, and allows finding them easily.

Reviewer 2: ```suggestion
Moreover, this EIP has a single structure that connects to all types of royalty types and allows finding them more easily.
```

Lastly, moving EIP-20 tokens around is much easier than managing an 0xsplits contract.

### Separating the work from the sellable NFT enables scaling licensing types

> By separating the canonical version of the work from its various licensed uses (NFT purchase, streaming, usage of art etc..) we are allowing a new ecosystem of various license types and payment types for a single work, based on the same standard.

Reviewer 2: ```suggestion
By separating the canonical version of the work from its various licensed uses (NFT purchase, streaming, usage of art etc..) this EIP introduces a new ecosystem of various license types and payment types for a single work, based on the same standard.
```

Author: I like the direction, but it is not introducing a new ecosystem, but introducing a root for an ecosystem to evolve. Makes sense?

In other words, when people use this schmeme, they won't start by creating a music NFT or art NFT, they start by creating the works registration, and then create types of licenses for it, each as its own sellable NFT, possible in the form of [EIP-5218](./eip-5218.md) or other formats.

### A single pointer to the works metadata

> The work points to metadata housed in IPFS or arweave and allows changing it and keeping track of the changes in a simple and standard way. Today the only metadata standard is NFT metadata extension, but we do not know which standard the document adheres to. With different work types we can forumlate different metadata standards for different work types and have a simple easy place to do discover attached metadata.

Reviewer 2: ```suggestion
The work points to metadata housed in IPFS or Arweave and allows changing it and keeping track of the changes in a simple and standard way. Today the only metadata standard is NFT metadata extension, but we do not know which standard the document adheres to. With different work types, different metadata standards for different work types can be formulated and have a simple easy place to discover attached metadata.
```

We foresee new marketplaces that allow registering your work, all base don the IWorksRegistration standard, but implemented for different types of works: Books, video, music and more. These marketplaces will support familiar royalty interest use cases, with the added benefits of transparent, fast and permissionless web3 infrastructure.

## Security Considerations

> There are no security considerations related directly to the implementation of this standard.

Reviewer: [peer review] I'd personally feel there shall be security considerations for distributing royalties, such as preventing reentry etc. But it's up to you to decide.
